test/library/idpdeployment.go

```diff
@@ -0,0 +1,209 @@
+
+	cleanup = func() {
+		// remove the NS, it will take away all the resources create here along with it
+		if err := clients.CoreV1().Namespaces().Delete(testContext, namespace, metav1.DeleteOptions{}); err != nil {
+			t.Logf("error cleaning up a resource: %v", err)
+		}
+	}
+
+	defer func() {
+		if err != nil {
+			cleanup()
+		}
+	}()
+
+	pod := podTemplate(name, image)
+	pod.Spec.Volumes = volumes
+	pod.Spec.Containers[0].VolumeMounts = volumeMounts
+	pod.Spec.Containers[0].Env = env
+	pod.Spec.Containers[0].Resources = resources
+	_, err = clients.CoreV1().Pods(namespace).Create(testContext, pod, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	_, err = clients.CoreV1().Services(namespace).Create(testContext, svcTemplate(), metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	route, err := routeClient.Routes(namespace).Create(testContext, routeTemplate(), metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	host, err = WaitForRouteAdmitted(t, routeClient, route.Name, route.Namespace)
+	require.NoError(t, err)
+
+	return
+}
+
+func podTemplate(name, image string) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"app": "e2e-tested-app",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "payload",
+					Image: image,
+					SecurityContext: &corev1.SecurityContext{
+						Privileged: boolptr(true),
+					},
+					Ports: []corev1.ContainerPort{
+						{
+							ContainerPort: 443,
+						},
+						{
+							ContainerPort: 80,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+func svcTemplate() *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "pod-svc",
+			Labels: CAOE2ETestLabels(),
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": "e2e-tested-app",
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name: "https",
+					Port: 443,
+				},
+				{
+					Name: "http",
+					Port: 80,
+				},
+			},
+		},
+	}
+}
+
+func routeTemplate() *routev1.Route {
+	return &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-route",
+		},
+		Spec: routev1.RouteSpec{
+			TLS: &routev1.TLSConfig{
+				Termination:                   routev1.TLSTerminationEdge,
+				InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
+			},
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: "pod-svc",
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromString("http"),
+			},
+		},
+	}
+}
+
+func CleanIDPConfigByName(t *testing.T, configClient configv1client.OAuthInterface, idpName string) {
+	config, err := configClient.Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("cleanup: failed to retrieve oauth/cluster: %v", err)
+	}
+
+	i := 0
+	for _, idp := range config.Spec.IdentityProviders {
+		if idp.Name == idpName {
+			break
+		}
+		i++
+	}
+
+	// did not find the idp of name
```

Comment:
(No action required) Where possible, consider using a boolean conditional whose name documents your intent in preference to a statement that requires a comment e.g.

```golang
notFound := i == len(config.Spec.IdentityProviders)
if notFound {
  return
}
```

Reply:
I usually stick just with comments to avoid unnecessary variable allocation

Reply:
I encourage you to avoid this kind of premature optimization. Code is for people first and foremost, and where optimization is required it generally requires performance analysis to determine bottlenecks (e.g. hot loops or stale references).

Reply:
Yet comments are there for the people and not the machine, and if the variable is not reused anywhere, a comment still makes more sense to me than a new variable